Coordinates in sanakoyeu_et_al_2018.RandomCrop.generate_anchors() are interchanged

```python
import torch
import pystiche_papers.sanakoyeu_et_al_2018 as paper

crop_width = 100
crop_height = 1

torch.manual_seed(0)
image_width = crop_width * 2
image_height = crop_height * 2

anchors = paper.RandomCrop.generate_anchors(
    batch_size=1,
    image_size=(image_height, image_width),
    crop_size=(crop_height, crop_width),
    same_on_batch=True,
)
anchor = anchors[0, 0, :]
print(anchor)

x, y = anchor.tolist()
assert x < image_width, f"{x} >= {image_width}"
assert y < image_height, f"{y} >= {image_height}"
```

```
tensor([ 0, 77], dtype=torch.int32)
AssertionError: 77 >= 2
```

This happens because the image and crop size place the height before the width (as is the default in PyTorch)

https://github.com/pmeier/pystiche_papers/blob/f47871f9334d21adc27de979c2480274bb73170e/pystiche_papers/sanakoyeu_et_al_2018/_data.py#L194-L200

whereas [`kornia.augmentation.functional.apply_crop`](https://kornia.readthedocs.io/en/latest/augmentation.html?highlight=apply_crop#kornia.augmentation.functional.functional.apply_crop) expects the horizontal coordinate before the vertical.

Cc @jbueltemeier

--- a/pystiche_papers/sanakoyeu_et_al_2018/_data.py
+++ b/pystiche_papers/sanakoyeu_et_al_2018/_data.py
@@ -197,7 +197,9 @@ class RandomCrop(AugmentationBase2d):
             )
             for image_length, crop_length in zip(image_size, crop_size)
         ]
-        return torch.stack(single_dim_anchors, dim=1,).unsqueeze(1).repeat(1, 4, 1)
+        return (
+            torch.stack(single_dim_anchors, dim=1).flip(1).unsqueeze(1).repeat(1, 4, 1)
+        )
 
     @staticmethod
     def clamp_vertices_to_size(

--- a/tests/unit/sanakoyeu_et_al_2018/test_data.py
+++ b/tests/unit/sanakoyeu_et_al_2018/test_data.py
@@ -26,7 +26,7 @@ from pystiche.misc import to_2d_arg
 from pystiche_papers.utils import make_reproducible
 
 from . import make_paper_mock_target
-from tests import asserts, parametrize
+from tests import asserts, mocks, parametrize
 from tests.utils import make_tar
 
 
@@ -139,11 +139,17 @@ def test_OptionalUpsample_repr(subtests):
             assert_property_in_repr(name, value)
 
 
-def test_RandomCrop_size(image):
-    edge_size = extract_edge_size(image) // 2
-    transform = paper.RandomCrop(edge_size)
+def test_RandomCrop_size(mocker, image_small_landscape):
+    mocker.patch(
+        mocks.make_mock_target("sanakoyeu_et_al_2018", "_data", "_adapted_uniform_int"),
+        side_effect=lambda shape, low, high, same_on_batch: torch.ones(
+            shape, dtype=torch.int32
+        ).mul(high),
+    )
+    edge_size = extract_edge_size(image_small_landscape) // 2
+    transform = paper.RandomCrop(edge_size, p=1.0)
 
-    output_image = transform(image)
+    output_image = transform(image_small_landscape)
     assert extract_image_size(output_image) == to_2d_arg(edge_size)
 
 

